### ISSUE TITLE
Compare nonlinear ES depth after sigma correction at equal 768 evaluations

### ISSUE THREAD

**Issue by Ahmadfauzi34** [Ahmadfauzi34/ai_agent_builder#236]
## Purpose

Follow #233/#234 and the newer hyperparameter evidence from #237/#238.

The original version of this issue planned to compare depth versus breadth at 768 evaluations using the old BASE OpenES settings (`sigma=0.15`, `learning_rate=0.06`). #238 materially changes that premise: at only 384 evaluations, `sigma=0.08` reached a substantially better final champion and much shorter plateau than BASE, and also outperformed the previous 768-evaluation baseline-budget result.

Therefore, do not spend the next experiment repeating the old sigma. Re-test depth versus breadth under the best validated one-variable-at-a-time search setting first.

## Fixed workload

Keep unchanged:

- fresh installed Python wheel and typed host surface;
- policy `Linear(6->8)->ReLU->Linear(8->2)`;
- canonical parameter dimension 74;
- same deterministic 2D dynamics, disturbances, four scenarios, reward/cost terms, action clamp, and 16-step horizon;
- OpenES strategy 0 and seed `20_260_917`;
- **sigma = 0.08** (best validated #238 OAT case);
- learning rate = 0.06;
- Python owns environment/reward/schedule, graph owns policy execution, ES owns candidate generation/update;
- `ask_f32()` candidate windows feed `apply_flat` directly.

Do not combine sigma and learning-rate tuning in this slice.

## Controlled matrix

Use independent optimizers from the same zero-policy baseline:

```text
C_LOW_SIGMA: population 8,  generations 48 = 384 evaluations
D_LOW_SIGMA: population 16, generations 48 = 768 evaluations
E_LOW_SIGMA: population 8,  generations 96 = 768 evaluations
```

Interpretation:

- C -> E isolates generation depth at fixed population under the corrected sigma.
- D vs E is the key equal-768 breadth-versus-depth comparison.
- C also anchors directly to the validated #238 S_LOW configuration.

## Required evidence

For every case report:

- zero-policy, initial champion, and final champion reward;
- baseline/training gain ratios;
- generation-best and champion histories;
- optimizer flags and exact candidate count;
- number of champion improvements;
- longest champion plateau;
- first/last champion-improvement generation;
- `NO_IMPROVEMENT` count;
- median ask/apply/rollout/tell timing and total wall time;
- finite fitness;
- stable batch cardinality;
- stable graph/program identity and binding identity.

For the highest validated final-reward case require exact stateful ProgramBundle parameter replay and reward replay within `1e-9`.

## Decision boundary

- E beats D at equal 768 evaluations -> generation depth remains the stronger lever after sigma correction; inspect depth saturation before optimizer algorithm changes.
- D beats E -> population breadth becomes more useful once exploration radius is corrected.
- E materially improves C -> extra depth still pays under the corrected search dynamics.
- C/E plateau similarly despite corrected sigma -> fixed OpenES dynamics may be reaching a real algorithm-level limit; gather algorithm evidence before changing optimizer semantics.
- nonlinear policy still remains materially worse than the established simple Linear policy -> task simplicity remains a strong explanation; do not widen model/activation surface for capability claims alone.

## Architectural boundary

Research-only. No change to:

- OpenES algorithm semantics or adaptive schedules;
- ABI/facade;
- graph/binding semantics;
- activation/model surface;
- Linear runtime;
- Math Program v1-v9;
- ProgramBundle schema;
- Resolution/Authorization;
- support matrix.

No NumPy/DLPack/PyO3 and no Go/C++ work.

Implement only after the duplicate budget gate cleanup in #239/#240 is merged so the next expensive research slice does not carry redundant CI.
